src/types/mod.rs

```diff
@@ -226,8 +227,36 @@ impl days_ms {
     }
 }
 
+const MS_IN_DAY: i32 = 86_400_000;
+
+impl Ord for days_ms {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let self_days = self.days() + self.milliseconds() / MS_IN_DAY;
+        let other_days = other.days() + other.milliseconds() / MS_IN_DAY;
+        let ord = self_days.cmp(&other_days);
+        match ord {
+            Ordering::Greater | Ordering::Less => ord,
+            Ordering::Equal => {
+                (self.milliseconds() % MS_IN_DAY).cmp(&(other.milliseconds() % MS_IN_DAY))
+            }
+        }
+    }
+}
+
+impl PartialOrd for days_ms {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for days_ms {
```

Comment:
```suggestion
impl Eq for days_ms {
```

The easiest way imo is to manually implement `Eq` (instead of `PartialEq`): `PartialEq` is automatically derived when an implementation of `Eq` exists.

I think that this is also valid for the `MonthsDaysNs`: in both cases reflexivity exists and thus we can implement `Eq` (instead of `PartialEq`).